tests/integration/targets/ec2_ami/tasks/main.yml

```diff
@@ -112,6 +112,53 @@
 
     # ============================================================
 
+    - name: create an image from the instance with attached devices with no_device true
+      ec2_ami:
+        name: '{{ ec2_ami_name }}_no_device_true_ami'
+        instance_id: '{{ setup_instance.instance_ids[0] }}'
+        device_mapping:
+          - device_name: /dev/sda1
+            volume_size: 10
+            delete_on_termination: true
+            volume_type: gp2
+          - device_name: /dev/sdf
+            no_device: yes
+        state: present
+        wait: yes
+        root_device_name: /dev/xvda
+      register: result_no_device_true
+
+    - name: set image id fact for deletion later
+      set_fact:
+        ec2_ami_no_device_true_image_id: "{{ result_no_device_true.image_id }}"
+
+    - name: create an image from the instance with attached devices with no_device false
+      ec2_ami:
+        name: '{{ ec2_ami_name }}_no_device_false_ami'
+        instance_id: '{{ setup_instance.instance_ids[0] }}'
+        device_mapping:
+          - device_name: /dev/sda1
+            volume_size: 10
+            delete_on_termination: true
+            volume_type: gp2
+            no_device: no
+        state: present
+        wait: yes
+        root_device_name: /dev/xvda
+      register: result_no_device_false
+
+    - name: set image id fact for deletion later
+      set_fact:
+        ec2_ami_no_device_false_image_id: "{{ result_no_device_false.image_id }}"
+
+    - name: assert that image with no_device option yes and no_device option no has been created
+      assert:
+        that:
+          - "result_no_device_true.changed"
```

Comment:
testing that there was a changed is good for what was expected in the issue, is there a way to test that the change is as expected ?? using ``ec2_ami_info``

Reply:
I checked that the result is as expected by  checking for the device mapping in the output recieved after creating ami rather than using the ec2_ami_info module to make it simpler and have fewer lines of code.